httpmw/operationid/operationid.go

```diff
@@ -0,0 +1,44 @@
+package operationid
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+type contextKey string
+
+const (
+	OperationIDKey   contextKey = "operationID"
+	CorrelationIDKey contextKey = "correlationID"
+	// http header name ACS RP adds for it's operation ID
+	RequestAcsOperationIDHeader = "x-ms-acs-operation-id"
+	// Details can be found here:
+	// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#client-request-headers
+	RequestCorrelationIDHeader = "x-ms-correlation-request-id"
```

Comment:
How about the `x-ms-client-request-id`?

Reply:
added headers from toolkit log as default, custom header extraction function can be used as needed